src/rgw/rgw_auth.cc

```diff
@@ -508,3 +508,44 @@ RGWAuthApplier::aplptr_t RGWKeystoneAuthEngine::authenticate() const
 
   return nullptr;
 }
+
+rgw::auth::Engine::result_t
+rgw::auth::Strategy::authenticate(const req_state* const s) const
+{
+  for (const stack_item_t& kv : auth_stack) {
+    const rgw::auth::Engine& engine = kv.first;
+    const auto& policy = kv.second;
+
+    auto&& res = engine.authenticate(s);
```

Comment:
> changing this to `auto res =` would incur an extra call to the move constructor

IIRC the standard explicitly allows (but not enforces) compilers to apply NVRO in such situation. So, depending on compiler, we can obtain the benefit without introducing the visual clutter of `auto&&`.